lib/matplotlib/figure.py

```diff
@@ -2084,6 +2084,216 @@ def tight_layout(self, renderer=None, pad=1.08, h_pad=None, w_pad=None,
+                if axc.xaxis.get_label_position() == labpo:
+                    ss = axc.get_subplotspec()
+                    nrows, ncols, rowc0, rowc1, colc, col1 = \
+                            ss.get_rows_columns()
+                    if (labpo == 'bottom') and (rowc1 == row1):
+                        same += [axc]
+                    elif (labpo == 'top') and (rowc0 == row0):
+                        same += [axc]
+
+            for axx in same:
+                _log.debug(' Same: %s', axx.xaxis.label)
+                axx.xaxis._align_label_siblings += [ax.xaxis]
+
+    def align_ylabels(self, axs=None, renderer=None):
+        """
+        Align the ylabels of subplots in this figure.
+
+        If a label is on the left, it is aligned with labels on axes that
+        also have their label on the left and that have the same
+        left-most subplot column.  If the label is on the right,
+        it is aligned with labels on axes with the same right-most column.
+
+        Parameters
+        ----------
+        axs : list of `~matplotlib.axes.Axes` (None)
+            Optional list of `~matplotlib.axes.Axes` to align
+            the ylabels.
+
+        renderer : (None)
+            Optional renderer to do the adjustment on.
+
+        See Also
+        --------
+        matplotlib.figure.Figure.align_xlabels
+
+        matplotlib.figure.Figure.align_labels
+
+        Example
+        -------
+        Example with large yticks labels::
+
+            fig, axs = plt.subplots(2, 1)
+            axs[0].plot(np.arange(0, 1000, 50))
+            axs[0].set_ylabel('YLabel 0')
+            axs[1].set_ylabel('YLabel 1')
+            fig.align_ylabels()
+
+        """
+
+        from .tight_layout import get_renderer
+
+        if renderer is None:
+            renderer = get_renderer(self)
+
+        if axs is None:
+            axs = self.axes
+
+        axs = np.asarray(np.array(axs)).flatten().tolist()
+        for ax in axs:
+            _log.debug(' Working on: %s', ax.get_ylabel())
+            ss = ax.get_subplotspec()
+            nrows, ncols, row0, row1, col0, col1 = ss.get_rows_columns()
+            same = [ax]
+            labpo = ax.yaxis.get_label_position()
+            for axc in axs:
+                if axc != ax:
+                    if axc.yaxis.get_label_position() == labpo:
+                        ss = axc.get_subplotspec()
+                        nrows, ncols, row0, row1, colc0, colc1 = \
+                                ss.get_rows_columns()
+                        if (labpo == 'left') and (colc0 == col0):
+                            same += [axc]
+                        elif (labpo == 'right') and (colc1 == col1):
+                            same += [axc]
+            for axx in same:
+                _log.debug(' Same: %s', axx.yaxis.label)
+                axx.yaxis._align_label_siblings += [ax.yaxis]
+
+    # place holder until #9498 is merged...
+    def align_titles(self, axs=None, renderer=None):
```

Comment:
I think this is independent of #9498 (on the feature side; the implementations may be overlapping...), or did I miss somthing?

Reply:
Oh, I remember now...  

This code (`align_titles`) is not needed now, and hence the call to it is commented out below, because titles are always in the same position relative to the top of the axis (if placed automatically).  So, going through this function is needlessly expensive (though for practical purposes its not that expensive).

If #9498 gets merged, the y-position of titles will be different, and this code will become useful.

No test, because if the title is placed manaully this shouldn't run....